### ISSUE TITLE
--wraps doesn't with for argument to cd

### ISSUE THREAD

**Issue by gleich** [fish-shell/fish-shell#9526]
fish version: fish, version 3.6.0
OS: Darwin gleich-2.wireless.rit.edu 22.3.0 Darwin Kernel Version 22.3.0: Thu Jan  5 20:48:54 PST 2023; root:xnu-8792.81.2~2/RELEASE_ARM64_T6000 arm64
Terminal: xterm-256color

## Problem

So I have a function called `conf` that simply just wraps around the cd command to .config:

```
function conf --wraps="cd /Users/matt/.config/$argv"
	cd /Users/matt/.config/$argv
end
```

When I press tab after typing out `conf` it only does auto-completion for the current working directory. Is there something that I am missing?

**Comment by mrmeszaros** [comment]
You need to add completions.

I am no fish-mage, but adding this after Your function definition should make it work:
```fish
function conf --wraps="cd ~/.config/$argv"
    cd ~/.config/$argv
end

complete --command conf  --exclusive --arguments "(ls -1 ~/.config)"
```
First one's on the house 😉

**PS**.: This is a _hacky_ solution - I suggest You follow the best practices:
- the function goes to `~/.config/fish/functions/conf.fish`
- the completions go to `~/.config/fish/completions/conf.fish`

**PPS**.: May I suggest You some friends that might help along Your journey:
- The Official Fish documentation: https://fishshell.com/docs/current/completions.html (pretty decent)
- Forums like https://StackOverflow.com and https://SuperUser.com should have a lot of Your possible issues already answered
- And lets not forget ye olde `man complete` as well as checking out other completions scripts.

**Comment by faho** [comment]
@mrmeszaros No, it's a bug - there are some special cases for cd in the completion system.

And you might mean well posting all those links, but:

1. This issue is a few months old, we just haven't gotten around to fixing it
2. It comes across as a tad condescending
>PPPS.: Unless this SHOULD work out of the box - this issue could be converted into a discussion I guess?

We don't typically do that, no. If there's a simple question that can just be answered it can also just be an issue that can be answered and closed.

(frankly I don't really see the point in github's discussions)

**Comment by mrmeszaros** [comment]
Hey, thanks for the clarification!

I didn't mean to be a know-it-all - I just hoped to provide information in a fun way, but I see how it could be interpreted otherwise.

Also thanks for marking it as a bug - it clarifies that this indeed is an issue with fish!

Third, I didn't know `--wraps` was so powerful to interpret the command and kind of evaluate it to see what should match `$argv`. Could You link the docs where it is explained? I couldn't seem to find anything mentioned about variable matching or `$argv`.
